module5/src/com/epam/dragon/logic/TreasureAction.java

```diff
@@ -0,0 +1,46 @@
+package com.epam.dragon.logic;
+
+import com.epam.dragon.entity.Treasure;
+import com.epam.dragon.starter.TaskLauncher;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TreasureAction {
+    public void showAllTreasures() {
+        for (Treasure treasure: TaskLauncher.dragon.getTreasures()) {
```

Comment:
Why do all these methods have access to your dragon which is in TaskLauncher? Why just not to pass dragon as a param?